src/compiler/checker.ts

```diff
@@ -28892,7 +28892,7 @@ namespace ts {
                 return errorType;
             }
             const facts = getTypeFacts(type);
-            if (facts & TypeFacts.IsUndefinedOrNull) {
+            if (facts & TypeFacts.IsUndefinedOrNull || type.flags & (TypeFlags.Intersection | TypeFlags.Instantiable) && (getBaseConstraintOfType(type) || unknownType) === unknownType) {

```

Comment:
I'm not super sure if this is the best check, I've based this on the logic from `getTypeFacts`:
https://github.dev/microsoft/TypeScript/blob/b379e7fc791fc8646d9d255adce5127a6ab83099/src/compiler/checker.ts#L23562-L23564

In fact, I think this case probably should report a similar error like the one introduced here:
https://github.com/microsoft/TypeScript/pull/49119/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R18815-R18817
Note that this logic has been further adjusted in:
https://github.com/microsoft/TypeScript/pull/49481/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R18795

cc @ahejlsberg